src/submission/submission.resolver.ts

```diff
@@ -0,0 +1,45 @@
+import { SubmissionService } from './submission.service';
+import { Resolver, Args, Mutation, ID, Query } from '@nestjs/graphql';
+import { UseGuards, NotFoundException } from '@nestjs/common';
+import { SubmissionEntity as Submission } from './entity/submission.entity';
+import { SubmissionDto } from './dto/submission.dto';
+import { GqlJwtAuthGuard } from 'src/common/guards/gql-jwt-auth.guard';
+import { EvaluationEvent } from './dto/evaluation-event.dto';
+
+@Resolver(() => Submission)
+export class SubmissionResolver {
+  constructor(private readonly submissionService: SubmissionService) {}
+
+  @Query(() => Submission)
+  @UseGuards(GqlJwtAuthGuard)
+  async submission(@Args({ name: 'id', type: () => ID }) id: string): Promise<Submission> {
+    const submission = await this.submissionService.getSubmission(id);
```

Comment:
Before returning, we should check either if the user owns the submission or is ADMIN... otherwise, the request shall not pass.